### DISCUSSION TITLE
Having trouble connecting to bloop: java.util.concurrent.TimeoutException

### DISCUSSION THREAD

**Discussion by msolomon-ck** [scalameta/metals#3507] [Q&A]
After a few attempts connection to bloop fails with the following error

```
java.util.concurrent.TimeoutException
	at java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1784)
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1928)
	at scala.meta.internal.metals.BuildServerConnection$.initialize(BuildServerConnection.scala:419)
	at scala.meta.internal.metals.BuildServerConnection$.$anonfun$fromSockets$1(BuildServerConnection.scala:329)
	at scala.util.Success.$anonfun$map$1(Try.scala:255)
	at scala.util.Success.map(Try.scala:213)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

I noticed that this happens when using the following java version

```
java -version
openjdk version "1.8.0_292"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_292-b10)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.292-b10, mixed mode)
```

Everything seems to work when using AdoptOpenJDK 11 but I have other worksheet related issues when using that version that look very similar to this issue https://github.com/scalameta/metals/issues/2187

Wondering if anyone else has bumped into this issue?

**Comment by tgodzik** [comment]
Thanks for the question, you should be able to see something more in the logs aside from the stacktrace, is there anything?

**Reply by passion123** [reply]
Hi ，i have the same error，and the complete log as below: could u give some adviece  ?


Java home: /usr/lib/jvm/java-8-openjdk-amd64/jre
Metals version: 0.11.12
2023.06.07 03:14:35 INFO  logging to file /home/project/Test/.metals/metals.log
2023.06.07 03:14:35 INFO  Started: Metals version 0.11.12 in workspace '/home/project/Test' for client ide
2023.06.07 03:14:35 INFO  time: initialize in 0.59s
Initializing Scala Debugger
2023.06.07 03:14:36 INFO  Attempting to connect to the build server...
2023.06.07 03:14:36 INFO  Bloop uses /usr/lib/jvm/java-8-openjdk-amd64/jre defined at /home/theia/.bloop/bloop.json
2023.06.07 03:14:35 INFO  skipping build import with status 'Failed'
2023.06.07 03:14:36 INFO  tracing is disabled for protocol BSP, to enable tracing of incoming and outgoing JSON messages create an empty file at /home/project/Test/.metals/bsp.trace.json or /home/theia/.cache/metals/bsp.trace.json
2023.06.07 03:14:37 INFO  no build target found for /home/project/Test/src/main/java/com/zgdz/VehiclePassVerificationScalaModel.scala. Using presentation compiler with project's scala-library version: 3.2.2
2023.06.07 03:14:40 INFO  time: code lens generation in 3.64s
Error, couldn't show release notes for Metals 0.11.12
Timeout occured during get request at https://api.github.com/repos/scalameta/metals/releases/tags/v0.11.12
2023.06.07 03:14:56 ERROR Timeout waiting for 'build/initialize' response
2023.06.07 03:14:56 WARN  Retrying connection to the build server Bloop
2023.06.07 03:14:56 INFO  Bloop uses /usr/lib/jvm/java-8-openjdk-amd64/jre defined at /home/theia/.bloop/bloop.json
2023.06.07 03:14:56 INFO  tracing is disabled for protocol BSP, to enable tracing of incoming and outgoing JSON messages create an empty file at /home/project/Test/.metals/bsp.trace.json or /home/theia/.cache/metals/bsp.trace.json
2023.06.07 03:15:16 ERROR Timeout waiting for 'build/initialize' response
2023.06.07 03:15:16 WARN  Retrying connection to the build server Bloop
2023.06.07 03:15:16 INFO  Bloop uses /usr/lib/jvm/java-8-openjdk-amd64/jre defined at /home/theia/.bloop/bloop.json
2023.06.07 03:15:16 INFO  tracing is disabled for protocol BSP, to enable tracing of incoming and outgoing JSON messages create an empty file at /home/project/Test/.metals/bsp.trace.json or /home/theia/.cache/metals/bsp.trace.json
2023.06.07 03:15:36 ERROR Timeout waiting for 'build/initialize' response
2023.06.07 03:15:36 WARN  Retrying connection to the build server Bloop
2023.06.07 03:15:36 INFO  Bloop uses /usr/lib/jvm/java-8-openjdk-amd64/jre defined at /home/theia/.bloop/bloop.json
2023.06.07 03:15:36 INFO  tracing is disabled for protocol BSP, to enable tracing of incoming and outgoing JSON messages create an empty file at /home/project/Test/.metals/bsp.trace.json or /home/theia/.cache/metals/bsp.trace.json
2023.06.07 03:15:56 ERROR Timeout waiting for 'build/initialize' response
2023.06.07 03:15:56 WARN  Retrying connection to the build server Bloop
2023.06.07 03:15:56 INFO  Bloop uses /usr/lib/jvm/java-8-openjdk-amd64/jre defined at /home/theia/.bloop/bloop.json
2023.06.07 03:15:56 INFO  tracing is disabled for protocol BSP, to enable tracing of incoming and outgoing JSON messages create an empty file at /home/project/Test/.metals/bsp.trace.json or /home/theia/.cache/metals/bsp.trace.json
2023.06.07 03:16:16 ERROR Timeout waiting for 'build/initialize' response
2023.06.07 03:16:16 WARN  Retrying connection to the build server Bloop
2023.06.07 03:16:16 INFO  Bloop uses /usr/lib/jvm/java-8-openjdk-amd64/jre defined at /home/theia/.bloop/bloop.json
2023.06.07 03:16:16 INFO  tracing is disabled for protocol BSP, to enable tracing of incoming and outgoing JSON messages create an empty file at /home/project/Test/.metals/bsp.trace.json or /home/theia/.cache/metals/bsp.trace.json
2023.06.07 03:16:36 ERROR Timeout waiting for 'build/initialize' response
2023.06.07 03:16:36 ERROR Failed to connect with build server, no functionality will work.
java.util.concurrent.TimeoutException
	at java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1784)
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1928)
	at scala.meta.internal.metals.BuildServerConnection$.initialize(BuildServerConnection.scala:530)
	at scala.meta.internal.metals.BuildServerConnection$.$anonfun$fromSockets$1(BuildServerConnection.scala:439)
	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:467)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)

2023.06.07 03:16:36 WARN  no build target for: /home/project/temp/src/main/java/com/test/Test.scala
